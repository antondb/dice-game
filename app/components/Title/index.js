/**
 *
 * Title
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledH1 = styled.h1`
  color: #f5f0f7;
  font-family: 'Anton', sans-serif;
  padding: 0.5em 1em;
`;

function Title(props) {
  const { children } = props;
  return <StyledH1>{children}</StyledH1>;
}

Title.propTypes = {
  children: PropTypes.node,
};

export default Title;
