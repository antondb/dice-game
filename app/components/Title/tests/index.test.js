/**
 *
 * Tests for Title
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Title from '../index';

describe('<Title />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<Title />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<Title />);
    expect(firstChild).toMatchSnapshot();
  });
});
