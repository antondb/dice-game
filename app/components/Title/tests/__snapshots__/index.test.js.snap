// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> Should render and match the snapshot 1`] = `
<h1
  class="Title__StyledH1-sc-1q7q8nk-0 BJgVs"
/>
`;
